client/lib/post-normalizer/rule-content-safe-images.js

```diff
@@ -1,160 +1,86 @@
-				parsedImgSrc = url.parse( imgSource, false, true );
-			}
-
-			safeSource = safeImageURL( imgSource );
-			if ( ! safeSource && parsedImgSrc.search ) {
-				// we can't make externals with a querystring safe.
-				// try stripping it and retry
-				parsedImgSrc.search = null;
-				parsedImgSrc.query = null;
-				safeSource = safeImageURL( url.format( parsedImgSrc ) );
-			}
-
-			removeUnsafeAttributes( image );
-
-			if ( ! safeSource || imageShouldBeRemovedFromContent( imgSource ) ) {
-				image.parentNode.removeChild( image );
-				// fun fact: removing the node from the DOM will not prevent it from loading. You actually have to
-				// change out the src to change what loads. The following is a 1x1 transparent gif as a data URL
-				image.setAttribute( 'src', TRANSPARENT_GIF );
-				image.removeAttribute( 'srcset' );
-				return;
-			}
+/** Checks whether or not imageUrl shoudl be removed from the dom
+ * @param {string} imageUrl - the url of the image
+ * @returns {boolean} whether or not it should be removed from the dom
+ */
+const imageShouldBeRemovedFromContent = ( imageUrl ) => {
+	if ( ! imageUrl ) {
+		return;
+	}
 
-			if ( maxWidth ) {
-				safeSource = maxWidthPhotonishURL( safeSource, maxWidth );
-			}
+	const bannedUrlParts = [
+		'feeds.feedburner.com',
+		'feeds.wordpress.com/',
+		'.feedsportal.com',
+		'wp-includes',
+		'wp-content/themes',
+		'wp-content/plugins',
+		'stats.wordpress.com',
+		'pixel.wp.com'
+	];
+
+	return some( bannedUrlParts, ( part ) => includes( imageUrl.toLowerCase(), part ) );
+};
+
+const makeImageSafe = ( post, image, maxWidth ) => {
+	let imgSource = image.getAttribute( 'src' );
+	const parsedImgSrc = url.parse( imgSource, false, true );
+	const hostName = parsedImgSrc.hostname;
+
+	// if imgSource is relative, prepend post domain so it isn't relative to calypso
+	if ( ! hostName ) {
+		imgSource = url.resolve( post.URL, imgSource );
+	}
 
-			image.setAttribute( 'src', safeSource );
+	const safeSource = maxWidthPhotonishURL( safeImageURL( imgSource ), maxWidth );
+	removeUnwantedAttributes( image );
 
-			if ( image.hasAttribute( 'srcset' ) ) {
-				let imgSrcSet;
-				try {
-					imgSrcSet = srcset.parse( image.getAttribute( 'srcset' ) );
-				} catch ( ex ) {
-					// if srcset parsing fails, set the srcset to an empty array. This will have the effect of removing the srcset entirely.
-					imgSrcSet = [];
-				}
-				imgSrcSet = imgSrcSet.map( imgSrc => {
-					if ( ! url.parse( imgSrc.url, false, true ).hostname ) {
-						imgSrc.url = url.resolve( post.URL, imgSrc.url );
-					}
-					imgSrc.url = safeImageURL( imgSrc.url );
-					return imgSrc;
-				} ).filter( imgSrc => imgSrc.url );
-				const newSrcSet = srcset.stringify( imgSrcSet );
-				if ( newSrcSet ) {
-					image.setAttribute( 'srcset', newSrcSet );
-				} else {
-					image.removeAttribute( 'srcset' );
-				}
-			}
+	// trickery to remove it from the dom / not load the image
+	// TODO: test if this is necessary
```

Comment:
specifically I was talking about testing if its necessary to change src to transparent gif for fakedom to not pre-fetch images